components/provider_ppucd/src/uniset_const.rs

```diff
@@ -0,0 +1,309 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/master/LICENSE ).
+#![allow(clippy::unreadable_literal, dead_code)]
+
+use crate::structs::*;
+use crate::support::*;
+use icu_provider::prelude::*;
+use icu_uniset::UnicodeSet;
+use std::borrow::Cow;
+use std::convert::TryInto;
+
+// helper fn
+fn get_prop(ppucd_provider: &PpucdDataProvider, resc_key: ResourceKey) -> UnicodeSet {
+    let data_req = DataRequest {
+        resource_path: ResourcePath {
+            key: resc_key,
+            options: ResourceOptions {
+                variant: None,
+                langid: None,
+            },
+        },
+    };
+    let mut resp: DataResponse<UnicodeProperty> = ppucd_provider.load_payload(&data_req).unwrap();
+
+    let ppucd_property_cow: Cow<UnicodeProperty> = resp.take_payload().unwrap();
+    let ppucd_property: UnicodeProperty = ppucd_property_cow.into_owned();
+    let uniset: UnicodeSet = ppucd_property.try_into().unwrap();
+    uniset
+}
+
+/// Upper
+pub fn upper(provider: &PpucdDataProvider) -> UnicodeSet {
```

Comment:
The signature of these functions should be

```rust
pub fn upper<'d, D: DataProvider<'d, UnicodeProperty<'d>> + ?Sized>(provider: &D) -> UnicodeSet
```

Note that PpucdDataProvider implements that trait.  However, doing this allows for other data provider implementations, such as FsDataProvider, which auto-implements that trait as long as UnicodeProperty is serializable.

Additionally, the functions should be in the uniset crate, not the provider_ppucd crate.

Reply:
Question about putting the functions in the uniset crate - wouldn't that create a dependency on PPUCD Provider from the UnicodeSet component? As it stands, there is a dependency in the reverse direction b/c parsing ppucd.txt required a data structure that is effectively UnicodeSet itself (contains a set of chars that be queried for inclusion). I figured that UnicodeSet should have minimal dependencies as a building block component, to the extent possible.

Reply:
It creates no such dependency! When you fix the function signatures to use the trait, you can just depend on icu_provider. You don't need to depend on ppucd.

Reply:
Oh, I should also mention that the struct definition also belongs in uniset.

Reply:
Done. Getter fns and structs moved to `props.rs` and `provider.rs` in the UnicodeSet crate, respectively.